### ISSUE TITLE
Travis deployment is broken

### ISSUE THREAD

**Issue by mboersma** [deis/docker-go-dev#32]
Tagged builds in master all have been [ending with](https://travis-ci.org/deis/docker-go-dev/builds/122891535):
```
Skipping a deployment with the script provider because this branch is not permitted
```

Does there need to be an `on: master` line in the `deploy` stanza to whitelist this branch explicitly? I couldn't tell from [the Travis docs](https://docs.travis-ci.com/user/deployment/).

(I published `quay.io/deis/go-dev:0.11.0` old-school, so this issue isn't blocking anything currently.)

**Comment by krancour** [comment]
@mboersma, builds from master always kick off a deploy.  If the sha on the head of master doesn't match a tag, the image that gets deployed is a mutable `latest`.  If, on the other hand, there is a tag matching that sha of the commit at the head of master, an immutable image with that semantic version is pushed...

The real problem here is that when we cut a release, the tag is usually applied in git well after the most recent CI build of master... and adding a tag doesn't currently trigger a new build.  (I'm not quite sure how to make that happen.)

Another workaround is to restart the most recent build of master _after_ the tag has been applied to the release in git.

**Comment by mboersma** [comment]
Closed by #38.
